src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java

```diff
@@ -47,6 +50,19 @@ private HttpSender getHttpSender() {
 
 	@Override
 	protected Void doInBackground() throws Exception {
+		TreeSet<HtmlParameter> cookies = this.httpMessage.getRequestHeader().getCookieParams();
+		if (targetToken.getType() == HtmlParameter.Type.cookie) {
+			Iterator<HtmlParameter> iter = cookies.iterator();
+			while (iter.hasNext()) {
+				HtmlParameter cookie = iter.next();
+				if (cookie.getName().equals(targetToken.getName())) {
+					iter.remove();
+					break;
+				}
+			}
+		}
+		HttpMessage msg = this.httpMessage.cloneRequest();
+		msg.getRequestHeader().setCookieParams(cookies);

```

Comment:
I will add a selection of this, however I would request of you to create a test case for the scenario that you are describing.